galsim/phase_psf.py

```diff
@@ -1169,18 +1162,29 @@ def __init__(self, screen_list, lam, t0=0.0, exptime=0.0, time_step=0.025, flux=
         # We'll set these more intelligently as needed below
         self._second_kick = second_kick
 
-        # Need to put in a placeholder SBProfile so that calls to, for example,
-        # self.stepk, still work.
-        array = np.array([[self._flux]], dtype=np.float)
-        bounds = galsim._BoundsI(1, 1, 1, 1)
-        wcs = galsim.PixelScale(self.scale)
-        image = galsim._Image(array, bounds, wcs)
-        dummy_interpolant = 'delta' # so wavefront gradient photon-shooting works.
-        self._dummy_obj = galsim.InterpolatedImage(
-                image, pad_factor=1.0, x_interpolant=dummy_interpolant,
-                _serialize_stepk=self._serialize_stepk,
-                _serialize_maxk=self._serialize_maxk)
-        self._sbp = self._dummy_obj._sbp
+        # Need to put in a placeholder SBProfile so that calls to stepk and maxk work
+        # without having to do _prepareDraw.
+        # All we really need is for the stepk and maxk to be correct, so use the
+        # force_ options to set them how we want.
+        if _force_stepk > 0.:
+            dummy_stepk = _force_stepk
+        else:
+            R = 0.5 * self.aper.diam / self.aper.pupil_plane_scale
+            R = (R**2 + 6**2)**0.5  # 6 = Quintic.xrange, which is the normal x_interpoalnt used.
+            R *= self.scale
+            dummy_stepk = np.pi / R
+        if _force_maxk > 0.:
+            dummy_maxk = _force_maxk
+        else:
+            dummy_maxk = self.aper._maxK(self.lam, self.scale_unit)
```

Comment:
For this one, I was actually copying the implementation of `nyquist_scale`.  This maxk produces an equivalent `nyquist_scale` as what that had been doing.  Agreed that it is probably larger than needed in many cases, but probably ok to leave it as is.  (Esp. since PhaseScreenList doesn't have a `_maxK` function.)